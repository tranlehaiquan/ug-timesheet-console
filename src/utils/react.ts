import * as React from 'react'

type IdentityFunction = <T>(fn: T) => T

export const typedMemo: IdentityFunction = React.memo

export const getDisplayName = <T extends any = any>(WrappedComponent: React.ComponentType<T>): string => {
  return WrappedComponent.displayName || WrappedComponent.name || 'Component'
}
