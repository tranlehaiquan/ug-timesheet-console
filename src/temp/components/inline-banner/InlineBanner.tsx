import * as React from "react";
import classnames from "classnames";

import { IconNames, Icon } from "../icon/Icon";

type BannerType = "general" | "success" | "warning" | "error";

export interface IInlineBannerProps
  extends React.HTMLAttributes<HTMLDivElement> {
  type: BannerType;
  /**
   * The type of banner. (general, success, warning or error)
   */
  className?: string;
}

interface InlineBannerStyleSet {
  icon: IconNames;
  classes: string;
  iconBackgroundClasses: string;
}

/**
 * Displays a inline banner with various styling depending on the type.
 */
export const InlineBanner: React.FC<IInlineBannerProps> = ({
  type,
  className,
  children,
  ...otherProps
}) => {
  const bannerStyleSet: { [name: string]: InlineBannerStyleSet } = {
    general: {
      classes: "sk-bg-blue-lighter",
      icon: "info",
      iconBackgroundClasses: "sk-bg-blue",
    },
    success: {
      classes: "sk-bg-green-lightest",
      icon: "tick",
      iconBackgroundClasses: "sk-bg-green",
    },
    error: {
      classes: "sk-bg-red-lightest",
      icon: "exclamation",
      iconBackgroundClasses: "sk-bg-red",
    },
    warning: {
      classes: "sk-bg-orange-lightest",
      icon: "exclamation",
      iconBackgroundClasses: "sk-bg-orange",
    },
  };

  const defaultClasses =
    "sk-text-navy sk-text-sm sk-font-medium sk-flex sk-items-center sk-w-full sk-py-3 sk-rounded";
  const { icon, classes, iconBackgroundClasses } = bannerStyleSet[type] || {
    classes: "",
    icon: null,
    iconBackgroundClasses: "",
  };

  return (
    <div
      {...otherProps}
      className={classnames(defaultClasses, classes, className)}
    >
      <div
        className={classnames(
          "sk-flex sk-justify-center sk-items-center sk-rounded-full sk-mx-3 sk-w-5 sk-h-5",
          iconBackgroundClasses
        )}
      >
        <Icon name={icon} size={10} className="sk-text-white" />
      </div>
      {children}
    </div>
  );
};
