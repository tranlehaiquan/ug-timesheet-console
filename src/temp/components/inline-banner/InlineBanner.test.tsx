import * as React from "react";
import { shallow } from "enzyme";

import { InlineBanner } from "./InlineBanner";

describe("InlineBanner", () => {
  test("success", () => {
    const component = shallow(
      <InlineBanner type="success">Inline banner</InlineBanner>
    );

    expect(component).toMatchSnapshot();
  });

  test("general", () => {
    const component = shallow(
      <InlineBanner type="general">Inline banner</InlineBanner>
    );

    expect(component).toMatchSnapshot();
  });

  test("warning", () => {
    const component = shallow(
      <InlineBanner type="warning">Inline banner</InlineBanner>
    );

    expect(component).toMatchSnapshot();
  });

  test("error", () => {
    const component = shallow(
      <InlineBanner type="error">Inline banner</InlineBanner>
    );

    expect(component).toMatchSnapshot();
  });
});
