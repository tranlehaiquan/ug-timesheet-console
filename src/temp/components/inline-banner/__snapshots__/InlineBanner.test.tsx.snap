// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineBanner error 1`] = `
<div
  className="sk-text-navy sk-text-sm sk-font-medium sk-flex sk-items-center sk-w-full sk-py-3 sk-rounded sk-bg-red-lightest"
>
  <div
    className="sk-flex sk-justify-center sk-items-center sk-rounded-full sk-mx-3 sk-w-5 sk-h-5 sk-bg-red"
  >
    <Icon
      className="sk-text-white"
      name="exclamation"
      size={10}
    />
  </div>
  Inline banner
</div>
`;

exports[`InlineBanner general 1`] = `
<div
  className="sk-text-navy sk-text-sm sk-font-medium sk-flex sk-items-center sk-w-full sk-py-3 sk-rounded sk-bg-blue-lighter"
>
  <div
    className="sk-flex sk-justify-center sk-items-center sk-rounded-full sk-mx-3 sk-w-5 sk-h-5 sk-bg-blue"
  >
    <Icon
      className="sk-text-white"
      name="info"
      size={10}
    />
  </div>
  Inline banner
</div>
`;

exports[`InlineBanner success 1`] = `
<div
  className="sk-text-navy sk-text-sm sk-font-medium sk-flex sk-items-center sk-w-full sk-py-3 sk-rounded sk-bg-green-lightest"
>
  <div
    className="sk-flex sk-justify-center sk-items-center sk-rounded-full sk-mx-3 sk-w-5 sk-h-5 sk-bg-green"
  >
    <Icon
      className="sk-text-white"
      name="tick"
      size={10}
    />
  </div>
  Inline banner
</div>
`;

exports[`InlineBanner warning 1`] = `
<div
  className="sk-text-navy sk-text-sm sk-font-medium sk-flex sk-items-center sk-w-full sk-py-3 sk-rounded sk-bg-orange-lightest"
>
  <div
    className="sk-flex sk-justify-center sk-items-center sk-rounded-full sk-mx-3 sk-w-5 sk-h-5 sk-bg-orange"
  >
    <Icon
      className="sk-text-white"
      name="exclamation"
      size={10}
    />
  </div>
  Inline banner
</div>
`;
